Userland/Libraries/LibCore/System.h

```diff
@@ -206,6 +210,27 @@ ErrorOr<void> unlockpt(int fildes);
 ErrorOr<void> access(StringView pathname, int mode);
 ErrorOr<DeprecatedString> readlink(StringView pathname);
 
+class VectorAndCustomDeleter {
+    AK_MAKE_NONCOPYABLE(VectorAndCustomDeleter);
+
+public:
+    VectorAndCustomDeleter(Vector<struct addrinfo>&& addresses, struct addrinfo* ptr, AK::Function<void(struct addrinfo*)> deleter)
```

Comment:
Sorry for reviewing separately again :sweat_smile: but this seems like a very generic name for a specific thing.
It's specifically specialized for `struct addrinfo`, so it seems like this is `AddressInfoList` or something like that (the current name seems to just describe an implementation detail).

Reply:
Don't worry, I often can't do it in one time either and it's still a review so thank you.

I will make these changes, but I can't do them tonight though